agent/api/task.go

```diff
@@ -251,10 +255,14 @@ func (task *Task) addNetworkResourceProvisioningDependency(cfg *config.Config) {
 		if container.IsInternal() {
 			continue
 		}
-		if container.SteadyStateDependencies == nil {
-			container.SteadyStateDependencies = make([]string, 0)
+		if container.TransitionDependencySet.ContainerDependencies == nil {
+			container.TransitionDependencySet.ContainerDependencies = make([]ContainerDependency, 0)
 		}
-		container.SteadyStateDependencies = append(container.SteadyStateDependencies, PauseContainerName)
+		container.TransitionDependencySet.ContainerDependencies = append(container.TransitionDependencySet.ContainerDependencies, ContainerDependency{
+			Container:       PauseContainerName,
+			SatisfiedStatus: ContainerRunning,
```

Comment:
If the `SatisfiedStatus ` means other containers can process after this container reached `SatisfiedStatus`, shouldn't this be `ContainerResourcesProvisioned`?

Reply:
`SatisfiedStatus` means that this particular dependent container can progress.  Satisfied doesn't necessarily mean `api.ContainerResourcesProvisioned`; it could also be `api.ContainerRunning` (and could be different statuses in the future).